x-pack/plugins/dataset_quality/public/hooks/use_dataset_quality_table.tsx

```diff
@@ -28,63 +28,83 @@ export const useDatasetQualityTable = () => {
   const {
     services: { fieldFormats },
   } = useKibanaContextForPlugin();
-  const [selectedDataset, setSelectedDataset] = useState<DataStreamStat>();
-  const [pageIndex, setPageIndex] = useState(0);
-  const [pageSize, setPageSize] = useState(10);
-  const [sortField, setSortField] = useState<SORT_FIELD>(DEFAULT_SORT_FIELD);
-  const [sortDirection, setSortDirection] = useState<DIRECTION>(DEFAULT_SORT_DIRECTION);
-
-  const defaultTimeRange = getDefaultTimeRange();
-
-  const { dataStreamsStatsServiceClient: client } = useDatasetQualityContext();
-  const { data = [], loading } = useFetcher(async () => client.getDataStreamsStats(), []);
-  const { data: degradedStats = [], loading: loadingDegradedStats } = useFetcher(
-    async () =>
-      client.getDataStreamsDegradedStats({
-        start: defaultTimeRange.from,
-        end: defaultTimeRange.to,
-      }),
-    []
+
+  const { service } = useDatasetQualityContext();
+
+  const { page, rowsPerPage, sort } = useSelector(service, (state) => state.context.table);
+
+  const flyout = useSelector(service, (state) => state.context?.flyout);
```

Comment:
**nit:** I think the optional change is not necessary here

Reply:
Addressed in [b4674a2](https://github.com/elastic/kibana/pull/174906/commits/b4674a2e466a9694051129f73ab366afa04551ab).